trickster/model.py

```diff
@@ -239,6 +238,87 @@ def select_response(self, responses: list[Response]) -> Response:
                 raise ValueError(f'Response selection algorithm for {self.value} is not configured.')
 
 
+class Auth(BaseModel):
```

Comment:
You should definitely create abstract methods `authenticate` and `from_dict` because you require them in the code later. Then it will be more apparent what is required to implement.

Reply:
I cant. Because Rout instantiate Auth field and gives error:
TypeError: Can't instantiate abstract class Auth with abstract method authenticate

Reply:
OK, then at least you should define signature of these methods here and within the body use `raise NotImplementedError('Implement this method in child class.')`